### PR TITLE
refactor!: remove deprecated core-enums, TuiColor, colorFallback, DEFAULT_COLORS, TuiColorHandler

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/themes/bootstrap/bootstrap.style.less [hunk 1]
```diff
@@ -36,7 +36,8 @@
             color: #959aa0;
         }
 
-        &[data-state='hovered'] {
+        &:hover:not(._no-hover),
```
Comment (waterplea) [inline projects/demo/src/modules/themes/bootstrap/bootstrap.style.less:39]: Same

File: projects/demo/src/modules/themes/material/material.style.less [hunk 2]
```diff
@@ -35,7 +35,8 @@
             transform: scale(0);
         }
 
-        &[data-state='hovered'] {
+        &:hover:not(._no-hover),
+        &[data-state='hover'] {
```
Comment (waterplea) [inline projects/demo/src/modules/themes/material/material.style.less:39]: And here

File: projects/kit/components/input-files/input-files.style.less [hunk 3]
```diff
@@ -41,15 +41,17 @@ tui-input-files {
         border: 1px solid;
     }
 
-    &[data-state='hovered'] {
+    &:hover:not(._no-hover),
+    &[data-state='hover'] {
```
Comment (waterplea) [inline projects/kit/components/input-files/input-files.style.less:45]: And here

**Review by waterplea — CHANGES_REQUESTED** [review]

File: projects/addon-preview/components/preview-dialog/preview-dialog.style.less [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
-    &[data-state='hovered'] {
-        background: rgba(159, 159, 159, 0.86);
-    }
-
-    &[data-state='pressed'] {
-        background: rgba(159, 159, 159, 0.75);
-    }
-
     &[data-state='disabled'] {
         opacity: var(--tui-disabled-opacity);
```
Comment (waterplea) [inline projects/addon-preview/components/preview-dialog/preview-dialog.style.less:15]: This is already defined in wrapper/base.less

File: projects/demo/src/modules/components/button/examples/4/index.less [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
-    }
+    });
 
-    &[data-state='pressed'] {
+    .wrapper-active({
         background: #8f75d1;
-    }
+    });
 
     &[data-state='disabled'] {
```
Comment (waterplea) [inline projects/demo/src/modules/components/button/examples/4/index.less:15]: I believe there's a mixin for that as well. Better to use those so it's centralized.

File: projects/demo/src/modules/themes/bootstrap/bootstrap.style.less [hunk 6]
```diff
@@ -36,9 +36,7 @@
             color: #959aa0;
         }
 
-        &[data-state='hovered'] {
-            background: transparent;
-        }
+        .wrapper-hover({background: transparent});
 
         &._focused:after {
```
Comment (waterplea) [inline projects/demo/src/modules/themes/bootstrap/bootstrap.style.less:41]: Focused state also has a mixin

File: projects/demo/src/modules/themes/material/material.style.less [hunk 7]
```diff
@@ -35,10 +35,10 @@
...
         }
 
-        &[data-state='hovered'] {
+        .wrapper-hover({
             background: #ececec;
             box-shadow: inset 0 -1px rgba(0, 0, 0, 0.87);
-        }
+        });
 
         &._focused {
```
Comment (waterplea) [inline projects/demo/src/modules/themes/material/material.style.less:43]: Same here. Let's use mixins whenever possible

File: projects/kit/components/input-files/input-files.style.less [hunk 8]
```diff
@@ -37,18 +38,6 @@ tui-input-files {
...
-        &:after {
-            color: var(--tui-link-hover);
-        }
-    }
-
-    &[data-state='pressed'] {
-        background: var(--tui-secondary-hover);
-    }
-
     &[data-state='disabled'] {
```
Comment (waterplea) [inline projects/kit/components/input-files/input-files.style.less:41]: Same here

**Review by waterplea — APPROVED** [review]

**Pull request merged**